### ISSUE TITLE
IPALLOC_RANGE improvement

### ISSUE THREAD

**Issue by tsunamireversal** [weaveworks/weave#3818]
** Is this a FEATURE REQUEST? **
yes, an improvement to allow automation software to easily find the tag to replace instead of trying to parse the weave.yaml file.

If so, please search existing feature requests, and if you find a similar one, up-vote it and/or add your comments to it instead. 

** why: **
As per https://www.weave.works/docs/net/latest/kubernetes/kube-addon/#configuration-options
in the section "Manually editing the YAML file", we unfortunately have to set this flag for many deployments, especially clouds like AWS and Azure, and/or whenever we hit a 10.x.x.x network (which is often) otherwise the host network clashes with the kubernetes (weave) one

** what **
What I would really like to see is either one of those two things to automate and simplify the flow:
*** Option 1 *** Add a comment just above where we would see IPALLOC_RANGE appear in, so it's findable by sed or ansible lineinfile module, such as:

      containers:
        - name: weave
          env:
            ### IPALLOC_RANGE

*** Option 2 *** Force it by default inside the file instead of making it optional, with an easy to recognize/modify subnet such as (just an example) 
      

      containers:
        - name: weave
          env:
            - name: IPALLOC_RANGE
              value: 172.209.0.0/16

To allow automation to replace it easily before applying it to a kubernetes master.
Also I believe a 172.209 (would still be a 16 bit network mask) and it would not clash out-of-the-box and could be applied to almost any configuration vs the current default of 10.x.x.x.

** applicability **
I believe this new feature should apply to all versions, unless I don't see something.

**Comment by bboreham** [comment]
Sounds reasonable.
Have you tried using a tool which is aimed at machine-editing yaml files like this, e.g. Kustomize?

**Comment by tsunamireversal** [comment]
Yes, I've used kustomize before, although not for this particular use case. I'd say not everyone uses kustomize (or any other tool that would solve this), as it introduces some new workflow and/or dependency to another tool in the equation.

I can continue and try to solve this using another parser, but really enjoyed the "it just works out-of-the-box" aspect which would work for (I believe) all network types, if the file was fixed at the source.

**Comment by bboreham** [comment]
>  I believe a 172.26 (would still be a 16 bit network mask) 

So 1/16th the size of the Weave Net default.

> and it would not clash 

Sadly, this is extremely unlikely.  It will clash with something.

This is exactly how we picked 10.32/12: we surveyed every cloud we had access to, and picked a range that didn't clash with any of them.

We can't change the default now: it would break everyone on upgrade.

**Comment by tsunamireversal** [comment]
I see, yeah, we don't want that, I guess I wrongly assumed it was a 16 bit network by default

Otherwise, can there be an anchor (comment) that would make it easy to add the IPALLOC_RANGE tag on the source file? That shouldn't break anything, right?

(option 1 above )

Thanks!

**Comment by tsunamireversal** [comment]
Any updates?

**Comment by bboreham** [comment]
Yes, adding a comment or the default to the yaml sounds reasonable.

**Comment by tsunamireversal** [comment]
Hi!
Just wondering... has it been implemented?

Thanks

**Comment by bboreham** [comment]
Nobody has contributed this change. 
Would you like to do it?

**Comment by tsunamireversal** [comment]
Sure! I'll send a contribution as soon as I can

**Comment by tsunamireversal** [comment]
Any suggestions?

**Comment by tsunamireversal** [comment]
Any pointers would be appreciated!
